plugins/modules/mso_schema_template_external_epg_selector.py

```diff
@@ -0,0 +1,244 @@
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['selector']],
+            ['state', 'present', ['selector']],
+        ],
+    )
+
+    schema = module.params.get('schema')
+    template = module.params.get('template')
+    external_epg = module.params.get('external_epg')
+    selector = module.params.get('selector')
+    expressions = module.params.get('expressions')
+    state = module.params.get('state')
+
+    mso = MSOModule(module)
+
+    # Get schema_id
+    schema_obj = mso.get_obj('schemas', displayName=schema)
+    if not schema_obj:
+        mso.fail_json(msg="Provided schema '{0}' does not exist".format(schema))
+
+    schema_path = 'schemas/{id}'.format(**schema_obj)
+
+    # Get template
+    templates = [t.get('name') for t in schema_obj.get('templates')]
+    if template not in templates:
+        mso.fail_json(msg="Provided template '{template}' does not exist. Existing templates: {templates}".format(template=template,
+                                                                                                                  templates=', '.join(templates)))
+    template_idx = templates.index(template)
+
+    # Get External EPG
+    externalEpgs = [e.get('name') for e in schema_obj.get('templates')[template_idx]['externalEpgs']]
+    if external_epg not in externalEpgs:
+        mso.fail_json(msg="Provided external epg '{externalEpg}' does not exist. Existing epgs: {externalEpgs}"
+                      .format(externalEpg=external_epg, externalEpgs=', '.join(externalEpgs)))
+    external_epg_idx = externalEpgs.index(external_epg)
+
+    # Get Selector
+    selectors = [s.get('name') for s in schema_obj.get('templates')[template_idx]['externalEpgs'][external_epg_idx]['selectors']]
+    if selector in selectors:
+        selector_idx = selectors.index(selector)
+        selector_path = '/templates/{0}/externalEpgs/{1}/selectors/{2}'.format(template, external_epg, selector_idx)
+        mso.existing = schema_obj.get('templates')[template_idx]['externalEpgs'][external_epg_idx]['selectors'][selector_idx]
+
+    if state == 'query':
+        if selector is None:
+            mso.existing = schema_obj.get('templates')[template_idx]['externalEpgs'][external_epg_idx]['selectors']
+        elif not mso.existing:
+            mso.fail_json(msg="Selector '{selector}' not found".format(selector=selector))
+        mso.exit_json()
+
+    selectors_path = '/templates/{0}/externalEpgs/{1}/selectors/-'.format(template, external_epg)
+    ops = []
+
+    mso.previous = mso.existing
+    if state == 'absent':
+        mso.existing = {}
+        ops.append(dict(op='remove', path=selector_path))
+
+    elif state == 'present':
+        # Get expressions
+        all_expressions = []
+        if expressions:
+            for expression in expressions:
+                IP = expression.get('type')
```

Comment:
don't use IP but prefer ip or ip_addr or ip_address as ALL_CAPS variable name are reserved for ENUMs or non mutable variables.